lib/rust/enso-prelude/src/vec.rs

```diff
@@ -0,0 +1,66 @@
+//! This module defines utilities for working with the [`std::vec::Vec`] type.
+
+
+// ==============
+// === VecOps ===
+// ==============
+
+pub trait VecOps {
+    type Item;
+
+    /// Pushes the provided `item` onto the [`std::vec::Vec`], and then returns an immutable
+    /// reference to the item.
+    fn push_and_get(&mut self, item:Self::Item) -> &Self::Item;
+
+    /// Pushes the provided `item` onto the [`std::vec::Vec`], and then returns a mutable reference
+    /// to the item.
+    fn push_and_get_mut(&mut self, item:Self::Item) -> &mut Self::Item;
+}
+
+impl <T> VecOps for Vec<T> {
+    type Item = T;
+
+    fn push_and_get(&mut self, item:Self::Item) -> &Self::Item {
+        self.push(item);
+        let item_ix = self.len() - 1;
+        self.get(item_ix).expect("Item should be presend as it has just been imported.")
```

Comment:
I believe you can use unrechable here. Probably no real performance benefit, but its more clean to me this wy tbh,